modules/core/src/va_intel.cpp

```diff
@@ -0,0 +1,517 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+// Copyright (C) 2015, Itseez, Inc., all rights reserved.
+// Third party copyrights are property of their respective owners.
+
+#include "precomp.hpp"
+
+#ifdef HAVE_VA
+#  include <va/va.h>
+#else  // HAVE_VA
+#  define NO_VA_SUPPORT_ERROR CV_ErrorNoReturn(cv::Error::StsBadFunc, "OpenCV was build without VA support (libva)")
+#endif // HAVE_VA
+
+using namespace cv;
+
+////////////////////////////////////////////////////////////////////////
+// CL-VA Interoperability
+
+#ifdef HAVE_OPENCL
+#  include "opencv2/core/opencl/runtime/opencl_core.hpp"
+#  include "opencv2/core.hpp"
+#  include "opencv2/core/ocl.hpp"
+#  include "opencl_kernels_core.hpp"
+#endif // HAVE_OPENCL
+
+#if defined(HAVE_VA_INTEL) && defined(HAVE_OPENCL)
+#  include <CL/va_ext.h>
+#endif // HAVE_VA_INTEL && HAVE_OPENCL
+
+namespace cv { namespace va_intel {
+
+#if defined(HAVE_VA_INTEL) && defined(HAVE_OPENCL)
+
+static clGetDeviceIDsFromVA_APIMediaAdapterINTEL_fn clGetDeviceIDsFromVA_APIMediaAdapterINTEL = NULL;
+static clCreateFromVA_APIMediaSurfaceINTEL_fn       clCreateFromVA_APIMediaSurfaceINTEL       = NULL;
+static clEnqueueAcquireVA_APIMediaSurfacesINTEL_fn  clEnqueueAcquireVA_APIMediaSurfacesINTEL  = NULL;
+static clEnqueueReleaseVA_APIMediaSurfacesINTEL_fn  clEnqueueReleaseVA_APIMediaSurfacesINTEL  = NULL;
+
+static bool contextInitialized = false;
+
+#endif // HAVE_VA_INTEL && HAVE_OPENCL
+
+namespace ocl {
+
+Context& initializeContextFromVA(VADisplay display, bool tryInterop)
```

Comment:
Can you clarify what is purpose for tryInterop param?
Can user just do not call initializeContextFromVA instead of call initializeContextFromVA(..,true) or Context::getDefault(false) is really required before start to use VA and UMat?
I see that in all other places after Context::getDefault(false) (with false arg) the initializeContextFromHandle is always called. Is it OK to call Context::getDefault(false) without initializeContextFromHandle call after it?